src/dft.cpp

```diff
@@ -174,7 +174,29 @@ dft_chunk *fields::add_dft(component c, const volume &where, const double *freq,
   dft_chunk_data data;
   data.c = c;
   data.vc = vc;
+
+  if (decimation_factor == 0) {
+    double src_freq_max = 0;
+    for (src_time *s = sources; s; s = s->next) {
+      if (s->get_width() == 0) {
+          decimation_factor = 1;
+        } else {
+        if (s->frequency().real()+0.5/s->get_width() > src_freq_max) {
```

Comment:
We really need a `get_fwidth(tol)` method that returns the bandwidth below which the Fourier transform decays below `tol`.

Reply:
(For most sources, `get_fwidth` will default to `0`.  For a Gaussian source we will use a formula, or maybe an approximate formula based on the continuous FT.   For adjoint sources, we have a formula for the Nutall windows.)

Reply:
Fixed. The function `gaussian_src_time::get_fwidth()` computes the bandwidth using an analytic formula for the continuous Fourier transform. The `get_fwidth()` function for the `FilteredSource` object used by the adjoint solver which involves a `CustomSource` will be added in a separate PR. In the meantime, the adjoint simulation does *not* use any decimation.